File: ProjectTest/OneLang/CSharp/One/Transforms/ResolveImports.cs
using One.Ast;

namespace One.Transforms
{
    public class ResolveImports {
        public static void processWorkspace(Workspace ws) {
            foreach (var pkg in Object.values(ws.packages))
                foreach (var file in Object.values(pkg.files))
                    foreach (var imp in file.imports) {
                        var impPkg = ws.getPackage(imp.exportScope.packageName);
                        var scope = impPkg.getExportedScope(imp.exportScope.scopeName);
                        imp.imports = imp.importAll ? scope.getAllExports() : imp.imports.map(x => x is UnresolvedImport unrImp ? scope.getExport(unrImp.name) : x);
                        file.addAvailableSymbols(imp.imports);
                    }
        }
    }
}

File: ProjectTest/OneLang/PHP/One/Transforms/ResolveImports.php
<?php

namespace One\Transforms\ResolveImports;

use One\Ast\Types\Workspace;
use One\Ast\Types\UnresolvedImport;

class ResolveImports {
    static function processWorkspace($ws) {
        foreach (array_values($ws->packages) as $pkg)
            foreach (array_values($pkg->files) as $file)
                foreach ($file->imports as $imp) {
                    $impPkg = $ws->getPackage($imp->exportScope->packageName);
                    $scope = $impPkg->getExportedScope($imp->exportScope->scopeName);
                    $imp->imports = $imp->importAll ? $scope->getAllExports() : array_map(function ($x) use ($scope) { return $x instanceof UnresolvedImport ? $scope->getExport($x->name) : $x; }, $imp->imports);
                    $file->addAvailableSymbols($imp->imports);
                }
    }
}
